test/ranking.jl

```diff
@@ -37,3 +38,69 @@ s = ["c", "a", "b", "d", "d", "b", "e", "d"] # s is a vector of strings ordered
 @test tiedrank(s) == tiedrank(x)
 @test tiedrank(x, rev = true) == tiedrank(-x)
 @test tiedrank(x, lt = (x, y) -> isless(y, x)) == tiedrank(-x)
+
+@testset "quantilerank and percentrank" begin
```

Comment:
Can you test corner cases where `v` is empty and of length 1? And when the value is outside the range of `v` (too small and too large)? It would also be good to test when `value` is the smallest or largest value in `v`, notably:
- when `v` has ties at the start and at the end
- when `v` doesn't have ties at the start nor at the end

Finally, inputs should not be sorted to make sure the code doesn't rely on that.

Reply:
I add some tests to fill those gaps you pointed. Mixing up some arrays to not be ordered.

here some few codes I wrote. 
```
@test_throws ArgumentError quantilerank([], 3)
@test_throws ArgumentError quantilerank([1], 3)
@test quantilerank(v4, -100, method=:inc) == 0.0
@test quantilerank(v4,  100, method=:inc) == 1.0
@test quantilerank(v4, -100, method=:exc) == 0.0
@test quantilerank(v4,  100, method=:exc) == 1.0
```